src/frontend/src/lib/components/ui/ButtonPrimary.svelte

```diff
@@ -0,0 +1,17 @@
+<script lang="ts">
+	import { createEventDispatcher } from 'svelte';
+
+	export let disabled = false;
+	export let onClick: (() => void) | undefined = undefined;
+
+	const dispatch = createEventDispatcher();
+
+	let handleOnClick = () => {
+		onClick?.();
+		dispatch('click');
+	};
+</script>
+
+<button class="primary block flex-1" {disabled} on:click={handleOnClick}>
```

Comment:
The type is missing, some buttons where `button` and some `submit`

Reply:
theoretically, that is already the default. You mean to explicitly show that we are going with the default?

Reply:
You are right and that was badly explained. Ultimately we will need both so let's already add the type as a property

```
export let type: "submit" | "button" = "submit";
```